firewood/src/merkle/node/partial_path.rs

```diff
@@ -0,0 +1,129 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE.md for licensing terms.
+
+use super::Flags;
+use crate::nibbles::NibblesIterator;
+use std::{
+    fmt::{self, Debug},
+    iter::once,
+};
+
+// TODO: use smallvec
+/// PartialPath keeps a list of nibbles to represent a path on the Trie.
+#[derive(PartialEq, Eq, Clone)]
+pub struct PartialPath(pub Vec<u8>);
+
+impl Debug for PartialPath {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> Result<(), fmt::Error> {
+        for nib in self.0.iter() {
+            write!(f, "{:x}", *nib & 0xf)?;
+        }
+        Ok(())
+    }
+}
+
+impl std::ops::Deref for PartialPath {
+    type Target = [u8];
+    fn deref(&self) -> &[u8] {
+        &self.0
+    }
+}
+
+impl From<Vec<u8>> for PartialPath {
```

Comment:
I don't like this addition. It's now easy for someone to put non-nibbles into a PartialPath. If you really think this is needed as a convenience, it should be TryFrom, and each byte should be checked.

This also ties PartialPath more strongly to a Vec, which is something I'd like to change.

Reply:
I created [#366](https://github.com/ava-labs/firewood/issues/366) suggesting to add convenience methods for extending and shrinking `PartialPath`. The current problem is that in order to perform any change to the path, you essentially need to copy the `PartialPath`, and then make changes to the internal `Vec` and reset the value. 

Adding `From` was a convenience for getting the rest of the "branch with path" implementation further along. I will improve this once it's complete.